src/scripting/logic.h

```diff
@@ -33,7 +33,8 @@ class LuaEditorInterface : public LuaInterface {
 	explicit LuaEditorInterface(Widelands::EditorGameBase* g);
 	~LuaEditorInterface() override = default;
 
-	std::unique_ptr<LuaTable> run_script(const std::string& script) override;
+	std::unique_ptr<LuaTable> run_script(const std::string& script,
+	                                     const bool no_pop_table = false) override;
```

Comment:
```/home/runner/work/widelands/widelands/src/scripting/logic.h:37:39: warning: parameter 'no_pop_table' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions [readability-avoid-const-params-in-decls]```
from run https://github.com/widelands/widelands/actions/runs/6782878008/job/18436004915?pr=6210#step:8:15

What do I have to do? I wrote `const` in logic.**h** and logic.**cc**, trying to do as I saw with the other parameters.

Reply:
This is a somewhat annoying check in clang-tidy. Just remove `const` in the header file but keep it in the .cc file.

Reply:
Oh, yes, just to make it even more confusing: `const X&` is a "const reference" to type `X` (note the `&` at the end), which is different from a normal reference (`X&`), so in that case you need the `const` in the header too.

Reply:
`no_pop_table` is a trivial type. Do not use const with trivial types such as int, bool. Those are copied anyway on function calls and changing them cannot have side effects.